File: site/swingjs/j2s/TwoDOF/Forced_2DOF/Forced/CFrame.js
(function(){var P$=Clazz.newPackage("TwoDOF.Forced_2DOF.Forced"),I$=[[0,'java.awt.Point','java.awt.Rectangle']],I$0=I$[0],$I$=function(i,n){return((i=(I$[i]||(I$[i]=Clazz.load(I$0[i])))),!n&&i.$load$&&Clazz.load(i,2),i)};
/*c*/var C$=Clazz.newClass(P$, "CFrame", null, 'java.awt.Rectangle');

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
},1);

C$.$fields$=[['Z',['mWasHit'],'O',['mFramePanel','TwoDOF.Forced_2DOF.Forced.CFramePanel']]]

Clazz.newMeth(C$, 'c$$TwoDOF_Forced_2DOF_Forced_CFramePanel',  function (thePanel) {
;C$.superclazz.c$$I$I$I$I.apply(this,[0, 0, 0, 0]);C$.$init$.apply(this);
this.mFramePanel=thePanel;
this.mFramePanel.AddFrame$TwoDOF_Forced_2DOF_Forced_CFrame(this);
}, 1);

Clazz.newMeth(C$, 'c$$TwoDOF_Forced_2DOF_Forced_CFramePanel$I$I$I$I',  function (thePanel, xx, yy, ww, hh) {
;C$.superclazz.c$$I$I$I$I.apply(this,[xx, yy, ww, hh]);C$.$init$.apply(this);
this.mFramePanel=thePanel;
this.mFramePanel.AddFrame$TwoDOF_Forced_2DOF_Forced_CFrame(this);
}, 1);

Clazz.newMeth(C$, 'Frame$java_awt_Graphics',  function (g) {
});

Clazz.newMeth(C$, 'MouseEvent$I$Z',  function (code, prevHit) {
this.mWasHit=this.contains$I$I(this.mFramePanel.mThisPt.x, this.mFramePanel.mThisPt.y);
return false;
});

Clazz.newMeth(C$, 'GlobalToLocal$java_awt_Point',  function (glob) {
return glob == null  ? null : Clazz.new_($I$(1,1).c$$I$I,[glob.x - this.x, glob.y - this.y]);
});

Clazz.newMeth(C$, 'GlobalToLocalR$java_awt_Rectangle',  function (glob) {
return glob == null  ? null : Clazz.new_($I$(2,1).c$$I$I$I$I,[glob.x - this.x, glob.y - this.y, glob.width, glob.height]);
});

Clazz.newMeth(C$, 'LocalToGlobal$java_awt_Point',  function (loc) {
return loc == null  ? null : Clazz.new_($I$(1,1).c$$I$I,[loc.x + this.x, loc.y + this.y]);
});

Clazz.newMeth(C$, 'LocalToGlobalR$java_awt_Rectangle',  function (loc) {
return loc == null  ? null : Clazz.new_($I$(2,1).c$$I$I$I$I,[loc.x + this.x, loc.y + this.y, loc.width, loc.height]);
});

Clazz.newMeth(C$, 'repaint$',  function () {
this.mFramePanel.repaint$();
});

Clazz.newMeth(C$, 'repaint$J',  function (delay) {
this.mFramePanel.repaint$J(delay);
});

Clazz.newMeth(C$, 'ControlMessage$I$D',  function (code, val) {
this.ControlMessage$TwoDOF_Forced_2DOF_Forced_CFrame$I$D(null, code, val);
});

Clazz.newMeth(C$, 'ControlMessage$TwoDOF_Forced_2DOF_Forced_CFrame$I$D',  function (controller, code, val) {
});

Clazz.newMeth(C$);
})();
;Clazz.setTVer('5.0.1-v7');//Created 2025-10-14 19:43:51 Java2ScriptVisitor version 5.0.1-v7 net.sf.j2s.core.jar version 5.0.1-v7

File: src/OneDOF/Prog5/CFrame.java
package OneDOF.Prog5.ForcedSin;

import java.awt.Graphics;
import java.awt.Point;
import java.awt.Rectangle;

@SuppressWarnings("serial")
class CFrame extends Rectangle {
   public static final int kMouseDown = 0;
   public static final int kMouseDrag = 1;
   public static final int kMouseUp = 2;
   public static final int kMouseMove = 3;
   public boolean mWasHit;
   CFramePanel mFramePanel;

   public CFrame(CFramePanel thePanel) {
      super(0, 0, 0, 0);
      this.mFramePanel = thePanel;
      this.mFramePanel.AddFrame(this);
   }

   public CFrame(CFramePanel thePanel, int xx, int yy, int ww, int hh) {
      super(xx, yy, ww, hh);
      this.mFramePanel = thePanel;
      this.mFramePanel.AddFrame(this);
   }

   public void Frame(Graphics g) {
   }

   public boolean MouseEvent(int code, boolean prevHit) {
      this.mWasHit = this.contains(this.mFramePanel.mThisPt.x, this.mFramePanel.mThisPt.y);
      return false;
   }

   public Point GlobalToLocal(Point glob) {
      return glob == null ? null : new Point(glob.x - super.x, glob.y - super.y);
   }

   public Rectangle GlobalToLocalR(Rectangle glob) {
      return glob == null ? null : new Rectangle(glob.x - super.x, glob.y - super.y, glob.width, glob.height);
   }

   public Point LocalToGlobal(Point loc) {
      return loc == null ? null : new Point(loc.x + super.x, loc.y + super.y);
   }

   public Rectangle LocalToGlobalR(Rectangle loc) {
      return loc == null ? null : new Rectangle(loc.x + super.x, loc.y + super.y, loc.width, loc.height);
   }

   public void repaint() {
      this.mFramePanel.repaint();
   }

   public void repaint(long delay) {
      this.mFramePanel.repaint(delay);
   }

   public void ControlMessage(int code, double val) {
      this.ControlMessage(null, code, val);
   }

   public void ControlMessage(CFrame controller, int code, double val) {
   }
}
